rust-src/id/src/test.rs

```diff
@@ -335,5 +394,370 @@ pub fn test_pipeline() {
+    assert!(
+        cdi_commitments.is_ok(),
+        "commitments Deserialization must be successful."
+    );
+
+    let cdi_proofs = serialize_deserialize(&cdi.proofs);
+    assert!(
+        cdi_proofs.is_ok(),
+        "Proof deserialization must be successful."
+    );
+
+    let des = serialize_deserialize(&cdi);
+    assert!(des.is_ok(), "Deserialization must be successful.");
+    // FIXME: Have better equality instances for CDI that do not place needless
+    // restrictions on the pairing (such as having PartialEq instnace).
+    // For now we just check that the last item in the proofs deserialized
+    // correctly.
+    assert_eq!(
+        des.unwrap().proofs.id_proofs.proof_reg_id,
+        cdi.proofs.id_proofs.proof_reg_id,
+        "It should deserialize back to what we started with."
+    );
+
+    // Revoking anonymity using all but one AR
+    let mut shares = Vec::new();
+    for (ar_id, key) in ars_secret.iter().skip(1) {
+        let ar = cdi
+            .values
+            .ar_data
+            .get(ar_id)
+            .expect(&format!("Anonymity revoker {} is not present.", ar_id));
+        let decrypted_share = (*ar_id, key.decrypt(&ar.enc_id_cred_pub_share));
+        shares.push(decrypted_share);
+    }
+    let revealed_id_cred_pub = reveal_id_cred_pub(&shares);
+    assert_eq!(
+        revealed_id_cred_pub,
+        global_ctx
+            .on_chain_commitment_key
+            .g
+            .mul_by_scalar(&id_use_data.aci.cred_holder_info.id_cred.id_cred_sec)
+    );
+
+    // generate a new cdi from a modified pre-identity object in which we swapped
+    // two anonymity revokers. Verification of this credential should fail the
+    // signature at the very least.
+    let (mut cdi, _) = create_credential(
+        context,
+        &id_object,
+        &id_use_data,
+        0,
+        policy,
+        &acc_data,
+        &SystemAttributeRandomness {},
+        &Left(EXPIRY),
+    )
+    .expect("Should generate the credential successfully.");
+    // Swap two ar_data values for two anonymity revokers.
+    let x_2 = cdi
+        .values
+        .ar_data
+        .get(&ArIdentity::new(2))
+        .expect("AR 2 exists")
+        .clone();
+    let x_3 = cdi
+        .values
+        .ar_data
+        .get(&ArIdentity::new(3))
+        .expect("AR 3 exists")
+        .clone();
+    *cdi.values
+        .ar_data
+        .get_mut(&ArIdentity::new(2))
+        .expect("AR 2 exists") = x_3;
+    *cdi.values
+        .ar_data
+        .get_mut(&ArIdentity::new(3))
+        .expect("AR 2 exists") = x_2;
+    // Verification should now fail.
+    let cdi_check = verify_cdi(&global_ctx, &ip_info, &ars_infos, &cdi, &Left(EXPIRY));
+    assert_ne!(cdi_check, Ok(()));
+}
+
+#[test]
+pub fn run_pipeline_old() { test_pipeline_old(); }
+
 #[test]
 pub fn run_pipeline() { test_pipeline(); }
+
+#[test]
+pub fn run_pipeline_v1() { test_pipeline_v1(); }
```

Comment:
Maybe we could get rid of these functions? They don't add value.

And add `#[test]` directly onto `test_pipeline` and `test_pipeline_v1`